src/main/scala/util/AsyncResetReg.scala

```diff
@@ -63,6 +63,9 @@ class AsyncResetRegVec(val w: Int, val init: BigInt) extends Module {
   }
 
   io.q := q.asUInt
+
+  override def desiredName = s"AsyncResetRegVec_w${w}_i${init}"
```

Comment:
Loving this change!

Reply:
This has been a known, but I think poorly advertised and utilized feature. We've discussed better ways of autogenerating such naming in Chisel based on a whitelist of parameter types. But I think good 'ole fashion style guides to implement `desiredName` for common library components will take us a very long way.